handlers/graph.go

```diff
@@ -145,11 +145,14 @@ func markTrafficGenerators(trafficMap graph.TrafficMap) {
 // buildNamespaceTrafficMap returns a map of all namespace nodes (key=id).  All
 // nodes either directly send and/or receive requests from a node in the namespace.
 func buildNamespaceTrafficMap(namespace string, o options.Options, client *prometheus.Client) graph.TrafficMap {
+	httpMetric := "istio_requests_total"
+	tcpMetric := "istio_tcp_sent_bytes_total"
+
 	// query prometheus for request traffic in three queries:
 	// 1) query for traffic originating from "unknown" (i.e. the internet)
-	groupBy := "source_workload_namespace,source_workload,source_app,source_version,destination_service_namespace,destination_service_name,destination_workload,destination_app,destination_version,response_code"
-	query := fmt.Sprintf("sum(rate(%v{reporter=\"destination\",source_workload=\"unknown\",destination_service_namespace=\"%v\",response_code=~\"%v\"} [%vs])) by (%v)",
-		o.Metric,
+	groupBy := "source_workload_namespace,source_workload,source_app,source_version,destination_workload_namespace,destination_service_name,destination_workload,destination_app,destination_version,response_code"
```

Comment:
I'm changing these queries to use `destination_workload_namespace` instead of `destination_service_namespace`. Sometimes, TCP timeseries doesn't have `destination_service_namespace` populated; looks safer to use the other one.

Reply:
-1 You absolutely can not change destination_service_namespace to destination_workload_namespace in the http request count queries.  This is critical to being able to catch source-side errors for requests that do not reach a destination workload.

As for TCP, I'm surprised that destination_service_namespace is not always set for reporter="source".  Do you have any examples of when this happens?  Perhaps it has to do with sent metrics as opposed to received metrics.  Or maybe the special query for unknown?

If you feel this is safe for TCP queries then OK, but don't change the HTTP handling.  And in general be very careful about changing existing logic.